src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java

```diff
@@ -346,40 +349,53 @@ public static final class QueueTaskCancelled extends CauseOfInterruption {
                 return;
             }
             LOGGER.fine(() -> "received node deletion event on " + node.getNodeName());
-            Timer.get().schedule(() -> {
-                Computer c = node.toComputer();
-                if (c == null || c.isOnline()) {
-                    LOGGER.fine(() -> "computer for " + node.getNodeName() + " was missing or online, skipping");
-                    return;
-                }
-                LOGGER.fine(() -> "processing node deletion event on " + node.getNodeName());
-                for (Executor e : c.getExecutors()) {
-                    Queue.Executable exec = e.getCurrentExecutable();
-                    if (exec instanceof PlaceholderTask.PlaceholderExecutable) {
-                        PlaceholderTask task = ((PlaceholderTask.PlaceholderExecutable) exec).getParent();
-                        TaskListener listener;
-                        try {
-                            listener = task.context.get(TaskListener.class);
-                        } catch (Exception x) {
-                            LOGGER.log(Level.WARNING, null, x);
-                            continue;
-                        }
-                        task.withExecution(execution -> {
-                            BodyExecution body = execution.body;
-                            if (body == null) {
-                                listener.getLogger().println("Agent " + node.getNodeName() + " was deleted, but do not have a node body to cancel");
-                                return;
-                            }
-                            listener.getLogger().println("Agent " + node.getNodeName() + " was deleted; cancelling node body");
-                            if (Util.isOverridden(BodyExecution.class, body.getClass(), "cancel", Throwable.class)) {
-                                body.cancel(new FlowInterruptedException(Result.ABORTED, false, new RemovedNodeCause()));
-                            } else { // TODO remove once https://github.com/jenkinsci/workflow-cps-plugin/pull/570 is widely deployed
-                                body.cancel(new RemovedNodeCause());
-                            }
-                        });
+            if (isOneShotAgent(node)) {
+                LOGGER.finest(() -> "Cancelling owner run for one-shot agent " + node.getNodeName() + " immediately");
```

Comment:
```suggestion
                LOGGER.fine(() -> "Cancelling owner run for one-shot agent " + node.getNodeName() + " immediately");
```
(This is relatively significant.)